### PR TITLE
Ensure synthetic source and dv codec are enabled with logs index mode.

### PR DESCRIPTION
**Pull request by martijnvg**
After running the elastic/logs track with logs index mode enabled, I noticed that _source was still getting stored. 
The issue was that other index modes than time_series weren't propagated to `Indexmetadata` and `IndexSettings` classes. Additionally the synthetic source defaults in `SourceFieldMapper` were geared towards time series index mode only. This PR addresses this.

### CODE REVIEW

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/8.15.0 / bwc-snapshots

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/bwc-snapshots

**Comment by martijnvg** [comment]
@elasticmachine run "elasticsearch-ci/8.15.0 / bwc-snapshots"

File: modules/aggregations/src/yamlRestTest/resources/rest-api-spec/test/aggregations/time_series.yml [hunk 1]
```diff
@@ -294,11 +294,11 @@ setup:
 ---
 "Configure with no synthetic source":
   - requires:
-      cluster_features: ["gte_v8.9.0"]
-      reason: "Error message fix in 8.9"
+      cluster_features: ["gte_v8.15.0"]
```
Comment (lkts) [inline modules/aggregations/src/yamlRestTest/resources/rest-api-spec/test/aggregations/time_series.yml:297]: You can use new capability @salvatore-campagna has introduced.

**Review by lkts — APPROVED** [review]

**Review by salvatore-campagna — APPROVED** [review]

**Comment by martijnvg** [comment]
@elasticmachine run tests

**Comment by martijnvg** [comment]
@elasticmachine update branch

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/bwc-snapshots

**Comment by martijnvg** [comment]
The 7.17.22 bwc test failures are unrelated (`Unknown NamedWriteable [org.elasticsearch.search.aggregations.PipelineAggregationBuilder]`) and are being tracked here: #109311.

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/bwc-snapshots

**Comment by benwtrent** [comment]
I am seeing a bunch of BWC failures in one of my PRs with the assertion tripping added in this PR.

Especially TSDB tests are failing, multiple bwc versions, etc.

An example: https://gradle-enterprise.elastic.co/s/nghp7qqiw7bzq

This reproduces locally: 
```
./gradlew ':qa:mixed-cluster:v8.2.3#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=tsdb/40_search/aggregate filter the tsid fails}" -Dtests.seed=D6CAB050D7FC5EA4 -Dtests.bwc=true -Dtests.locale=hi -Dtests.timezone=America/Godthab -Druntime.java=22
```

The failure of which crashes a node, so, any other tests that run against that testing cluster fails after. 

I am not sure which of the TSDB tests are effected. I would recommend reverting this and opening it again, but with `full-bwc` testing enabled.

**Pull request merged**